### PR TITLE
Pst 119 121

### PR DESCRIPTION
**Pull request by endynguyen**
[PST119]
Description:
Added API to create new product
Endpoint:
api/v1/products (post)

Request body:  
```
{
  "name": "Test product",
  "code": "TES",
  "sellingPrice": 20000,
  "categoryCode": "T1",
  "historicalPrice": 10000,
  "discountPrice": 15000,
  "description": "SDFDSNFLKG",
  "displayOrder": 2,
  "type": "SINGLE"
}
```
Response:
Product Id

[PST121]
Description:
Added API to view list product
Endpoint:
api/v1/products (get)

Response:
List product

### CODE REVIEW

File: Pos-System/Controllers/ProductController.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Pos_System.API.Constants;
+using Pos_System.API.Enums;
+using Pos_System.API.Payload.Request.Products;
+using Pos_System.API.Payload.Response.Products;
+using Pos_System.API.Services.Implements;
+using Pos_System.API.Services.Interfaces;
+using Pos_System.API.Validators;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
```
Comment (megastar24577) [inline Pos-System/Controllers/ProductController.cs:14]: Please delete this comment

File: Pos-System/Controllers/ProductController.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    public class ProductController : BaseController<ProductController>
+    {
+        private readonly IProductService _productService;
+        public ProductController(ILogger<ProductController> logger, IProductService productService) : base(logger)
+        {
+            _productService = productService;
+        }
+
+        // POST api/v1/products
+        [HttpPost]
```
Comment (megastar24577) [inline Pos-System/Controllers/ProductController.cs:28]: Remove this one

File: Pos-System/Controllers/ProductController.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            {
+                _logger.LogInformation(
+                    $"Create new product failed: {createNewProductRequest.Name}, {createNewProductRequest.Code}");
+                return Ok(MessageConstant.Product.CreateNewProductFailedMessage);
+            }
+
+            return Ok(response);
+        }
+
+        // GET: api/v1/products
```
Comment (megastar24577) [inline Pos-System/Controllers/ProductController.cs:46]: Please remove this comment

File: Pos-System/Payload/Response/Products/GetProductResponse.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Pos_System.API.Payload.Response.Products
+{
+    public class GetProductResponse
+    {
+        public Guid Id { get; set; }
+        public string Code { get; set; }
+        public string Name { get; set; }
+        public string? PicUrl { get; set; }
+        public Enums.ProductStatus Status { get; set; }
```
Comment (megastar24577) [inline Pos-System/Payload/Response/Products/GetProductResponse.cs:14]: Remove Enums.

File: Pos-System/Controllers/ProductController.cs (hunk 2 above)
Comment (endynguyen) [reply Pos-System/Controllers/ProductController.cs:26]: done. thank you

File: Pos-System/Pos-System.API.csproj [hunk 5]
```diff
@@ -7,6 +7,7 @@
     <RootNamespace>Pos_System.API</RootNamespace>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" '$(RunConfiguration)' == 'Pos_System_Backend' " />
```
Comment (megastar24577) [inline Pos-System/Pos-System.API.csproj:10]: Please don't change anything in this file

File: Pos-System/Properties/launchSettings.json [hunk 6]
```diff
@@ -1,4 +1,4 @@
-﻿{
+{
```
Comment (megastar24577) [inline Pos-System/Properties/launchSettings.json:1]: Please don't change this file either

**Pull request closed without merging**